tests/sdk/nodejs/nodejs_test.go

```diff
@@ -1338,3 +1338,145 @@ func TestServiceAccountTokenSecret(t *testing.T) {
 	})
 	integration.ProgramTest(t, &test)
 }
+
+// TestClientSideDriftCorrectCSA tests that we can successfully reapply a resource that has been
+// modified outside of Pulumi using the client-side apply strategy.
+func TestClientSideDriftCorrectCSA(t *testing.T) {
+	var ns, cmName string
+
+	applyStep := baseOptions.With(integration.ProgramTestOptions{
+		Dir:                  filepath.Join("drift-correct", "configmap-csa"),
+		ExpectRefreshChanges: true,
+		// Enable destroy-on-cleanup so we can shell out to kubectl to make external changes to the resource and reuse the same stack.
+		DestroyOnCleanup: true,
+		OrderedConfig: []integration.ConfigValue{
+			{
+				Key:   "pulumi:disable-default-providers[0]",
+				Value: "kubernetes",
+				Path:  true,
+			},
+		},
+		ExtraRuntimeValidation: func(t *testing.T, stackInfo integration.RuntimeValidationStackInfo) {
+			cm, ok := stackInfo.Outputs["cm"].(map[string]interface{})
+			assert.True(t, ok)
+
+			// Save the name and namespace for later use with kubectl. We check that the vars are empty,
+			// in case pulumi up creates a new ConfigMap/Namespace instead of updating the existing one on
+			// subsequent runs.
+			if ns == "" && cmName == "" {
+				ns = cm["metadata"].(map[string]interface{})["namespace"].(string)
+				cmName = cm["metadata"].(map[string]interface{})["name"].(string)
+			}
+
+			// Validate we applied ConfigMap with data.
+			fooV, ok, err := unstructured.NestedString(cm, "data", "foo")
+			assert.True(t, ok)
+			assert.NoError(t, err)
+			assert.Equal(t, "bar", fooV)
+		},
+	})
+
+	// Use manual lifecycle management since we need to run external commands in between pulumi up steps, while referencing
+	// the same stack.
+	pt := integration.ProgramTestManualLifeCycle(t, &applyStep)
+	err := pt.TestLifeCycleInitAndDestroy()
+	assert.NoError(t, err)
+
+	// Sanity check with kubectl to verify that the ConfigMap was created with the wanted label.
+	out, err := exec.Command("kubectl", "get", "configmap", "-o", "yaml", "-n", ns, cmName).CombinedOutput()
+	assert.NoError(t, err)
+	assert.Contains(t, string(out), "bar") // ConfigMap should have been created with data foo: bar.
+
+	// Update the ConfigMap and change the data foo: bar to foo: baz.
+	out, err = exec.Command("kubectl", "patch", "configmap", "-n", ns, cmName, "-p", `{"data":{"foo":"baz"}}`).CombinedOutput()
```

Comment:
I just checked, and `kubectl patch` does not update the `lastAppliedConfiguration` annotation, which is used for CSA diffing. See https://github.com/pulumi/pulumi-kubernetes/issues/694 for a related issue.

Reply:
Only `kubectl apply` attaches that annotation iirc. This bug would also be triggered by out of band resource modifications by other controllers/tools that use client-go, or other libraries.